src/lib/frontend/cnf.ml

```diff
@@ -239,9 +239,13 @@ and make_form up_qv name_base ~toplevel f loc ~decl_kind : E.t =
           let res = make_term up_qv name_base t in
           if negated then E.neg res else res
 
-        | TAneq lt | TAdistinct lt ->
+        | TAneq lt ->
           let lt = List.map (make_term up_qv name_base) lt in
           E.mk_distinct ~iff:true lt
+
+        | TAdistinct lt ->
+          E.mk_distinct ~iff:true (List.map (make_term up_qv name_base) lt)
+
```

Comment:
Can we avoid these changes? They don't seem useful, and keeping the patch as tiny as possible makes it easier to cherry-pick and move around if needed.

Reply:
`TAneq [x; y]` is the constructor used for `x <> y`. "neq" is a fairly common shorthand for "**n**ot **eq**ual".